### PR TITLE
Enable streaming option in the OpenAI API server

### LINKED ISSUE #459: Is openai compatible server still working?
In a fresh conda env, ater pip install, the following command (mentioned in this [PR](https://github.com/microsoft/DeepSpeed-MII/pull/317)) is no longer working.

```
python -m mii.entrypoints.openai_api_server     --model "mistralai/Mistral-7B-Instruct-v0.1"     --port 3000     --host 0.0.0.0
```

There are at least problems:

* Extra deps are not included, but needed: `shortuuid`, `uvicorn`, ...
* pydantic related. See logs below.

```
python -m mii.entrypoints.openai_api_server     --model "mistralai/Mistral-7B-Instruct-v0.1"     --port 3000     --host 0.0.0.0
[2024-04-15 15:55:20,541] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to cuda (auto detect)
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/workspace/envs/deepspeed/lib/python3.11/site-packages/mii/entrypoints/openai_api_server.py", line 31, in <module>
    from .data_models import (
  File "/workspace/envs/deepspeed/lib/python3.11/site-packages/mii/entrypoints/data_models.py", line 12, in <module>
    from pydantic import BaseModel, BaseSettings, Field
  File "/workspace/envs/deepspeed/lib/python3.11/site-packages/pydantic/__init__.py", line 380, in __getattr__
    return _getattr_migration(attr_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/envs/deepspeed/lib/python3.11/site-packages/pydantic/_migration.py", line 296, in wrapper
    raise PydanticImportError(
pydantic.errors.PydanticImportError: `BaseSettings` has been moved to the `pydantic-settings` package. See https://docs.pydantic.dev/2.7/migration/#basesettings-has-moved-to-pydantic-settings for more details.

For further information visit https://errors.pydantic.dev/2.7/u/import-error

```

### PR DESCRIPTION
**Pull request by adk9**
Now that token streaming support has merged (#397), we can enable streaming response in the OpenAI RESTful API endpoint.

This PR
* adds missing package dependencies for OpenAI API server (fixes #459)
* re-enables streaming responses for the OpenAPI API endpoint

#### Running the Server
```
python -m mii.entrypoints.openai_api_server \
    --model "mistralai/Mistral-7B-Instruct-v0.1" \
    --port 3000 \
    --host 0.0.0.0
```

#### Client
```
from openai import OpenAI

client = OpenAI(
    base_url="http://ip:port/v1",
    api_key="test",
)

completion = client.chat.completions.create(
    model="mistralai/Mistral-7B-v0.1",
    messages=[
        {
            "role": "user",
            "content": "Tell me a joke.",
        },
    ],
    max_tokens=1024,
    stream=True
)

for chunk in completion:
    if chunk.choices[0].delta.content is not None:
        print(chunk.choices[0].delta.content, end="")
```

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -2,13 +2,16 @@ accelerate
...
 deepspeed>=0.14.0
 deepspeed-kernels
+fastapi
+fastchat
 Flask-RESTful
 grpcio
 grpcio-tools
 Pillow
-pydantic
+pydantic==1.*
```
Comment (loadams) [inline requirements/requirements.txt:11]: FYI @adk9 - we're trying to get the [Pydantic update](!423) merged, hopefully very soon.  So I was curious on what issues you ran into that you needed to pin this?

Comment (adk9) [reply requirements/requirements.txt:11]: > FYI @adk9 - we're trying to get the [Pydantic update](https://github.com/microsoft/DeepSpeed-MII/pull/!423) merged, hopefully very soon.

Oh great, thanks for the pointer! I was just starting to look into this myself.

> So I was curious on what issues you ran into that you needed to pin this?

I see a `400 Bad Request` error. 
```
Error code: 400 - {'object': 'error', 'message': "[{'type': 'missing', 'loc': ('query', 'request'), 'msg': 'Field required', 'input': None}]", 'code': 40001}
```
I think the issue is with fastapi not fully supporting working with pydantic.v1 while having pydantic v2 installed (https://github.com/tiangolo/fastapi/issues/10360). So the workaround is to either pin fastapi or pydantic to an older version, and I chose to do the latter.

Comment (loadams) [reply requirements/requirements.txt]: Resolved as we have updated to pydantic 2

File: .github/workflows/nv-a6000-fastgen.yml [hunk 2]
```diff
@@ -35,8 +35,9 @@ jobs:
           python -c "import torch; print('CUDA available:', torch.cuda.is_available())"
       - name: Install transformers
         run: |
-          git clone --depth=1 https://github.com/huggingface/transformers
```
Comment (loadams) [inline .github/workflows/nv-a6000-fastgen.yml:38]: Revert these changes, as they are only required for now in order while we wait for this PR to be merged which fixes transformers for torch<2.4

Comment (loadams) [reply .github/workflows/nv-a6000-fastgen.yml]: PR to HF transformers completed.

**Review by loadams — APPROVED** [review]
